usecases/scenario_testrun_usecase.go

```diff
@@ -149,3 +149,37 @@ func (usecase *ScenarioTestRunUsecase) GetTestRunById(ctx context.Context,
 	}
 	return testrun, nil
 }
+
+func (usecase *ScenarioTestRunUsecase) CancelTestRunById(ctx context.Context,
+	testRunId string,
+) (models.ScenarioTestRun, error) {
+	exec := usecase.executorFactory.NewExecutor()
+	testRun, err := usecase.repository.GetTestRunByID(
+		ctx,
+		exec,
+		testRunId)
+	if err != nil {
+		return models.ScenarioTestRun{}, err
+	}
+	if err := usecase.enforceSecurity.ReadTestRun(testRun.OrganizationId); err != nil {
+		return models.ScenarioTestRun{}, err
+	}
+	if testRun.Status == models.Down {
+		return testRun, nil
+	}
+	if testRun.Status == models.Up {
```

Comment:
I don't think you need to handle the case "up" separately.
Anything that's not "down" should just be made to "down"